ADD BE UDF bitmap_and  and bitmap_or

DanyBin opened the pull request:
1. Add BE udf bitmap_and and bitmap_or
2.Test environment, modify production bitmap

Commit:
ADD BE BitMap UDF and & or

Commit:
ADD BE BitMap UDF and & or

Commit:
ADD BE BitMap UDF and & or

Commit:
ADD BE BitMap UDF and & or

Commit:
ADD BE BitMap UDF and & or

yangzhg committed:
[Index] Implements create drop show index syntax for bitmap index [#2487] (#2573)

### create table with index 
```
CREATE TABLE table1
(
    siteid INT DEFAULT '10',
    citycode SMALLINT,
    username VARCHAR(32) DEFAULT '',
    pv BIGINT SUM DEFAULT '0',
    INDEX index_name [USING BITMAP] (siteid, citycode) COMMENT 'balabala'
)
AGGREGATE KEY(siteid, citycode, username)
DISTRIBUTED BY HASH(siteid) BUCKETS 10
PROPERTIES("replication_num" = "1");
```
### create index 
```
CREATE INDEX index_name  ON table1 (siteid, citycod) [USING BITMAP] COMMENT 'balabala';
or 
ALTER TABLE table1 ADD  INDEX index_name  [USING BITMAP] (siteid, citycod) COMMENT 'balabala';
```
### drop index
```
DROP INDEX index_name ON table1;
or
ALTER TABLE table1 DROP INDEX index_name 
```

### show index
```
SHOW INDEX[ES] FROM table1
```
output
```
+---------+-------------+-----------------+------------+---------+
| Table   | Index_name  | Column_name     | Index_type | Comment |
+---------+-------------+-----------------+------------+---------+
| table1  | index_name  | siteid,citycode | BITMAMP    | balabala|
+---------+-------------+-----------------+------------+---------+
```

kangkaisen committed:
Fix HLL_UNION_AGG AnalyticFn result in BE core by adding hll_get_value (#2653)

kangkaisen committed:
Fix BITMAP_UNION_COUNT couldn't hit rollup table (#2655)

caiconghui committed:
Add filter conditions for 'show partitions from table' syntax (#2553)

Add filter conditions for show partitions from table syntax, to filter partitions needed

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,23 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
     return result;
 }
 
+StringVal BitmapFunctions::bitmap_or(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
```
src and dst can be null, you should handle this case.

DanyBin replied:
OK. I will update doc and fix this case.

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,23 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
 }
 
+StringVal BitmapFunctions::bitmap_or(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
+    RoaringBitmap src_bitmap ((char*)src.ptr);
```
src and dst can be already deserialized, you should handle the different cases.

imay commented on be/test/exprs/bitmap_function_test.cpp:
```diff
@@ -29,18 +29,16 @@
 namespace doris {
 
 StringVal convert_bitmap_to_string(FunctionContext* ctx, RoaringBitmap& bitmap) {
```
```suggestion
StringVal convert_bitmap_to_string(FunctionContext* ctx, const RoaringBitmap& bitmap) {
```

imay reviewed (changes requested):
should update documents for new added functions

caiconghui committed:
Fix doris be compile error for Ubuntu14.04 (#2647)

WingsGo committed:
[Dynamic Partition] Support for automatically adding partitions

In some scenarios, when a user creates an olap table that is range partition by time, the user needs to periodically add and remove partitions to ensure that the data is valid. As a result, adding and removing partitions dynamically can be very useful for users.

imay committed:
Adapt arrow 0.15 API (#2657)

This CL supports arrow's zero copy read interface, which can make code
comply with arrow 0.15.
And the schema change unit test has some problem, I disable it in run-ut.sh

WingsGo committed:
[Unit Test]Fix Schema Change Test Case (#2659)

Commit:
Merge branch 'be_bitmap_udf-0.1' of github.com:DanyBin/incubator-doris into be_bitmap_udf-0.1

kangkaisen committed:
Fix FE couldn't start (#2662)

imay committed:
Fix bug: parquet scanner don't seek (#2661)

WingsGo committed:
[Build]Make set target arch universal (#2660)

Commit:
fix bitmap udf

Commit:
fix bitmap udf

Commit:
fix bitmap udf

blueChild commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+    bitmap.serialize((char*)result.ptr);
+    return result;
+}
+StringVal BitmapFunctions::bitmap_and(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
```
is the third parameter called ‘dst’ appropriate?

Commit:
a

songenjie committed:
Fix bug: CreateIndexClause can be casted to AlterTableClause (#2667)

yangzhg committed:
Fix incompatibility with arm architecture in util and gutil (#2650)

1. upgrade  gutil code from imapla  to new verison， include `cpuinfo`, `spinlock` and `linux_syscall_support `
2. impliments arm version  utf8 check code
3. remove incompatible code from stopwatch

infearOnTheWay committed:
Add nio support for mysql protocol implementation (#2603)

Commit:
Merge branch 'be_bitmap_udf-0.1' of github.com:DanyBin/incubator-doris into be_bitmap_udf-0.1

Commit:
doc
